tests/generate/test_generate_client.py

```diff
@@ -70,3 +71,22 @@ def test_npx_failed(proc_mock: ProcMock, application_json: pathlib.Path) -> None
 
     assert result.exit_code == 1
     verify(result.output.replace("npx.cmd", "npx"))
+
+
+def test_snake_case() -> None:
+    assert snake_case("SnakeCase") == "snake_case"
+    assert snake_case("snakeCase") == "snake_case"
+    assert snake_case("snake-case") == "snake_case"
+    assert snake_case("snake_case") == "snake_case"
+    assert snake_case("SNAKE_CASE") == "snake_case"
+    assert snake_case("Snake_Case") == "snake_case"
+
+
+def test_format_client_name(application_json: pathlib.Path) -> None:
```

Comment:
these look good, just realized it might be useful to add some additional test cases to the `test_generate_client_python` and `test_generate_client_typescript` tests that use these tokens. Can use `@pytest.mark.parametrize` decorators to parameterize the output file name and expected output locations